server/src/main/java/io/crate/analyze/RoleAnalyzer.java

```diff
@@ -21,40 +21,59 @@
 
 package io.crate.analyze;
 
+import static io.crate.role.Role.Properties.JWT_KEY;
+import static io.crate.role.Role.Properties.PASSWORD_KEY;
+
+import java.util.Map;
+import java.util.Set;
+
 import io.crate.analyze.expressions.ExpressionAnalysisContext;
 import io.crate.analyze.expressions.ExpressionAnalyzer;
 import io.crate.analyze.relations.FieldProvider;
+import io.crate.expression.symbol.Literal;
 import io.crate.expression.symbol.Symbol;
 import io.crate.metadata.CoordinatorTxnCtx;
 import io.crate.metadata.NodeContext;
-import io.crate.sql.tree.AlterRole;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import io.crate.sql.tree.AlterRoleReset;
+import io.crate.sql.tree.AlterRoleSet;
 import io.crate.sql.tree.CreateRole;
 import io.crate.sql.tree.Expression;
 import io.crate.sql.tree.GenericProperties;
 
 public class RoleAnalyzer {
 
     private final NodeContext nodeCtx;
+    private final SessionSettingRegistry sessionSettingRegistry;
 
-    RoleAnalyzer(NodeContext nodeCtx) {
+    RoleAnalyzer(NodeContext nodeCtx, SessionSettingRegistry sessionSettingRegistry) {
```

Comment:
`SessionSettingRegistry` is now no longer used

Reply:
yep, thought I needed it for the CREATE ROLE validation but we check only the 2 props jwt and password.